solutions/gold/usaco-1019.mdx

```diff
@@ -0,0 +1,96 @@
+---
+id: usaco-1019
+source: USACO Gold 2020 February
+title: Delegation
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_deleg_gold_feb20.html)
+
+## Explanation
+
+First note that if $K$ is not a divisor of the $N - 1$, then the edges can
+never be evenly partitioned. Thus, we only need to check divisors of $N - 1$.
+
+We can arbitrarily root the tree at node $1$. Let $\mathtt{toBeMerged[i]}$ store the length of the path passes through node $i$ and its length has not reached $K$ yet. Since only $1$ of such path exist, all of the other paths which source from the subtrees of $i$ must be evenly paired.
+
+For each node $j$ such that $j$ is a child of node $i$, there must be another node $k$ such that $\mathtt{toBeMerged[j]+toBeMerged[k]+2} = K$. If no such $k$ exist, then we will store the current length of the path in $\mathtt{toBeMerged[i]}$, but there can only be one such $j$.
+
+Since the number of divisors for $N$ is around $\log N$, we will be iterating over all nodes only $\log N$ times.
```

Comment:
sauce?

Reply:
![image](https://github.com/cpinitiative/usaco-guide/assets/79723780/f0e502a2-7cc2-42de-8949-02569c77def1)

Reply:
i assume you're actually going to put a link instead of just talking to me here